talisker/celery.py

```diff
@@ -88,18 +103,29 @@ def after_task_publish(sender, body, **kwargs):
         talisker.statsd.get_client().timing(name, ms)
 
 
+def send_run_metric(name, ts):
+    name = 'celery.{}.run'.format(name)
+    ms = (time.time() - ts) * 1000
+    talisker.statsd.get_client().timing(name, ms)
+
+
 def task_prerun(sender, task_id, task, **kwargs):
-    name = 'celery.{}.run'.format(sender.name)
-    task.talisker_timer = talisker.statsd.get_client().timer(name)
-    task.talisker_timer.start()
     rid = get_header(task.request, REQUEST_ID)
     if rid is not None:
         talisker.request_id.push(rid)
+    if hasattr(task, 'talisker_timestamp') and task.request.is_eager:
+        # eager task, but retry has happened immeadiately, so send metrics
+        send_run_metric(sender.name, task.talisker_timestamp)
```

Comment:
yeah, that's the goal. As when eager, all preruns are fired before any postruns are called.

So the result of this PR is that in that case, the 2nd+ preruns send metrics for the retry, and the first postrun sends it for the final retry. The other postruns with then fire do nothing.

It's awkward, but it's what celery does, so..